packages/doenetml-worker-rust/lib-doenetml-core/src/core/general_state_var/string_state_var.rs

```diff
@@ -158,3 +158,273 @@ impl StateVarUpdater<String, StringRequiredData> for StringStateVar {
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn string_state_var_with_no_dependencies() {
+        // set up data that consists of just the essential variable
+        let mut string_data = StringRequiredData::default();
+        let essential = StateVarMutableView::new_with_value("".to_string(), true);
+        string_data.essential = essential.create_new_read_only_view();
+
+        // add meta data needed for invert to work. Use arbitrary values just to test.
+        string_data._data_query_mapping_data.essential.push((5, 7));
+
+        // a call to calculate gives the default "", as that is what essential contains
+        assert_eq!(
+            StringStateVar::calculate(&string_data),
+            StateVarCalcResult::FromDefault("".to_string())
+        );
+
+        // if change the value of essential, get calculated value with new value
+        essential.set_value("hello".to_string());
+        assert_eq!(
+            StringStateVar::calculate(&string_data),
+            StateVarCalcResult::Calculated("hello".to_string())
+        );
+
+        // if change the value of essential as though it were the default,
+        // get the new default value
+        essential.set_value_from_default("bye".to_string());
+        assert_eq!(
+            StringStateVar::calculate(&string_data),
+            StateVarCalcResult::FromDefault("bye".to_string())
```

Comment:
Maybe it's not good for our specific case, but could we use something like https://stackoverflow.com/questions/55079070/how-to-accept-str-string-and-string-in-a-single-function to remove all these `.to_string()`s?

Reply:
I don't think it helps as most of these are cases where the function needs to take ownership of the `String`. Could use `String::from("hi")` is that looks more pleasing!

Reply:
What if a `to_owned()` is called afterwards? Can we eat the reference?

Reply:
If one used the trait bound `AsRef<str>`, I don't think you can call `.to_owned()`, as I can't imagine `AsRef<str>` would imply an implementation of `.to_owned()`. My guess is the trait bound `AsRef<str>` is good only if all you need is a reference.